text/0141-linking-traces.md

```diff
@@ -0,0 +1,361 @@
+- Trace Duration: As soon as a trace spans more than one root span or multiple events, the duration of the trace itself becomes meaningless. While we strongly recommend to generally avoid deducing information from the trace duration, we could improve the current situation by sending less long-lived traces
+- Context: While long lived traces do potentially provide more context, given they could include spans or events that happened before an error, they can sometimes contain "too much" context. For instance, very long-running traces in web applications without navigations would contain close to the entire user journey. This makes it harder to understand cause-effect relationships within a trace.
+- UI/UX of the trace view: Long running traces are not ideally displayed in the trace view, given the x-axis would span a large time frame. This makes it necessary for users to zoom in and pan around much more than for single-root span traces.
+- Sampling and quota increase: If a trace is sampled positively, all transactions within this trace will be sampled positively. This also means that while the positively sampled trace is ongoing, we'd always propagate the positive sampling decision via the `sentry-trace` header to downstream services on outgoing requests. Assuming these services are configured to only sample already sampled traces, we'd send more events on an org-wide basis than if we rolled the dice more often (like in short-lived traces). This [increases quota usage](https://github.com/getsentry/sentry-ruby/issues/2318) for our customers without them necessarily being aware of it. 
+ 
+If we can link traces, we can ultimately shorten individual trace lengths and remove most of these disadvantages again.
+
+To be clear, the Sentry product will always need to support long-lived/multiple root span traces. While we can remove them from SDKs supporting this model, older SDK versions will stick around and users can also manually create long-lived traces.
+
+# Options Considered
+
+## [Preferred] Span Links
+
+We propose to add links on a span level, as defined and specified by [OpenTelemetry](https://opentelemetry.io/docs/concepts/signals/traces/#span-links). In addition to linking to span ids, a span link also holds meta information about the link, collected via attributes. We'll make use of these attributes by annotating our created span links with a `sentry.link.type` attribute which we can later use in the product to query for linked traces.
+
+### SDK APIs
+
+To support adding span links, SDKs need to expose at least an `addLink` method on their respective `Span` interface. For completeness with OpenTelemetry, ideally they also expose `addLinks`:
```

Comment:
just to mention: for the js sdk this is already exposed on the span interface but basically does nothing

Reply:
Good point, I'll add a note in the RFC. Also, I think this serves as a pro for going with span links: we already expose the API, and it not being functional is confusing for users familiar with pure Otel

Reply:
https://github.com/getsentry/rfcs/pull/141/commits/4cba46c50743a701d6862338b269b73e1fd3b73a